statsd/tag_cardinality.go

```diff
@@ -0,0 +1,78 @@
+package statsd
+
+import (
+	"os"
+	"strings"
+	"sync"
+)
+
+type Parameter interface{}
+
+// Possible values for the cardinality are "none", "low", "orchestrator", and "high".
+type CardinalityParameter struct {
+	card string
```

Comment:
Given that cardinality is a limited number of options, really we should use more of an enum for this. Go doesn't have proper enums but I think something like this would make the intent clearer. Instead of doing string processing to handle the cardinality we just have integer operations, so slightly more efficient:

```go
type Cardinality int

const (
	None Cardinality = iota
	Low
	Orchestrator
	High
)

func (c Cardinality) String() string {
	switch c {
	case None:
		return "none"
	case Low:
		return "low"
	case Orchestrator:
		return "orchestrator"
	case High:
		return "high"
	}

	return ""
}
```